### PR TITLE
CountByField support

### PR DESCRIPTION
**Pull request by u007**
example:
```
query.CountByField(&User{}, "distinct user_id")
```

### CODE REVIEW

File: finders.go [hunk 1]
```diff
@@ -133,6 +133,19 @@ func (q Query) Count(model interface{}) (int, error) {
 	return res.Count, err
 }
 
+func (q Query) CountByField(model interface{}, field string) (int, error) {
```
Comment (markbates) [inline finders.go]: This has almost the same code as `Count`. Can you update `Count` to call `CountByField` under the covers and pass `*` as the field?

**Comment by u007** [comment]
done :)

**Comment by markbates** [comment]
@u007 can you please rebase this with master? There are a lot of conflicts. Thanks.

**Comment by markbates** [comment]
Other than that this looks good to go!

**Comment by u007** [comment]
Hi Markbates,
it states: This branch has no conflicts with the base branch
ive done a pull from your master branch, but no changes. i've already did this before making a pull request. Did i miss anything?

**Comment by markbates** [comment]
@u007 it will let me do a merge commit, not a rebase. no matter, the merge will be fine. thanks.

**Comment by u007** [comment]
okay, i think i need to do a fresh fork of your master in the future

**Pull request merged**